src/erlcloud_aws.erl

```diff
@@ -494,3 +501,189 @@ get_filtered_statuses(ServiceNames, Statuses) ->
+                      session_secs = SessionDuration }.
+
+profiles_read() ->
+    HOME = profiles_home(),
+    Path = HOME ++ "/.aws/credentials",
+    case file:read_file( Path ) of
+        {ok, Contents} ->
+            profiles_parse( Contents );
+        Error ->
+            error_msg( "could not read credentials file ~s, because ~p",
+                   [Path, Error] )
+    end.
+    
+profiles_home() ->
+    case os:getenv("HOME") of
+        false ->
+            error_msg( "HOME environment variable is not set" );
+        HOME -> HOME
+    end.
+        
+profiles_parse( Content ) ->
+    case eini:parse( Content ) of
+        {ok, Profiles} -> Profiles;
+        Error ->
+            error_msg( "failed to parse credentials, because: ~p", [Error] )
+    end.
+            
+profiles_resolve( Name, Profiles, Options ) ->
+    profiles_resolve( Name, Profiles, undefined, Options ).
+
+profiles_resolve( BinName, Profiles, Role, Options ) when is_binary(BinName) ->
+    try binary_to_existing_atom( BinName, latin1 ) of
+        Name -> profiles_resolve( Name, Profiles, Role, Options )
+    catch
+        error:badarg -> 
+            error_msg( "invalid source_profile reference to ~s", [BinName] )
+    end;
+profiles_resolve( Name, Profiles, BinRole, Options ) when is_binary(BinRole) ->
+    Role = binary_to_list( BinRole ),
+    profiles_resolve( Name, Profiles, Role, Options );
+profiles_resolve( Name, Profiles, Role, Options ) ->
+    case proplists:get_value( Name, Profiles ) of
+        undefined ->
+            error_msg( "profile ~s does not exist", [Name] );
+        Keys ->
+            profiles_recurse( Keys, Profiles, Role, Options )
+    end.
+
+profiles_recurse( Keys, Profiles, Role, Options ) ->
+    case profiles_credentials( Keys ) of
+        {ok, Id, Secret} ->
+            profiles_assume( Id, Secret, Role, Options );
+        {cont, ProfileName, BinRole} ->
+            profiles_resolve( ProfileName, Profiles, BinRole, Options )
+    end.
+
+profiles_credentials( Keys ) ->
+    Names = [aws_access_key_id, aws_secret_access_key, source_profile],
+    BinRole = proplists:get_value( role_arn, Keys ),
+    case [proplists:get_value( K, Keys ) || K <- Names] of
+        [Id, Secret, undefined] when Id =/= undefined, Secret =/= undefined ->
+            {ok, binary_to_list(Id), binary_to_list(Secret)};
+        [undefined, undefined, BinProfile] when BinProfile =/= undefined ->
+            {cont, BinProfile, BinRole}
+    end.
+
+profiles_assume( Id, Secret, undefined, _Options ) ->
+    Config = #aws_config{ access_key_id = Id, secret_access_key = Secret },
+    {ok, Config};
+profiles_assume( Id, Secret, Role,
+                 #profile_options{ session_name = Name,
+                                   session_secs = Duration } ) ->
+    Config = #aws_config{ access_key_id = Id, secret_access_key = Secret },
+    {AssumedConfig, _Creds} =
+        erlcloud_sts:assume_role( Config, Role, Name, Duration, undefined ),
```

Comment:
same applies to `role_session_name`. this parameter is optional. Since we are trying to be on par with CLI I think we should supprt that too. 
see http://docs.aws.amazon.com/cli/latest/topic/config-vars.html